Objects/typevarobject.c

```diff
@@ -458,6 +461,73 @@ unpack_typevartuples(PyObject *params)
     }
 }
 
+static PyObject *
+typeparam_reduce_anonymous(PyObject *self, PyObject *owner)
+{
+    assert(PyTuple_Check(owner));
+    assert(PyTuple_GET_SIZE(owner) == 3);
```

Comment:
Is it possible to call that reduce function directly by calling some `__reduce__` with an illegal number of items here? (probably not as they should likely be generated by Python, but I'm just asking to be sure).

If we know that no evil Python code can trigger this call, you may want to assert with `CheckExact` (same for `PyUnicode_CheckExact(module_name)` instead)

Reply:
`__module__` and `__qualname__` can be set to arbitrary objects. Even though in this case it will not happen due to when `owner` is currently set, that may not always be the case in the future if the owner setting intrinsic is reused in some other way. Illegal number of items though no because that is set in this code itself. The size check is essentially free though so why not just keep the safety?

Reply:
> The size check is essentially free though so why not just keep the safety?

I'm not saying we need to remove them. What I'm saying is that "is it possible to craft an owner that is neither a tuple nor has 3 items and be able to call that function?" because on DEBUG builds, an assertion failure means a crash and we want to avoid this.

Reply:
Not currently, the tuple comes from the bytecode created by the compiler:
```
+ 13:13-13:14            LOAD_GLOBAL              0 (__name__)
+ 13:13-13:14            LOAD_CONST               1 ('cls.func')
+ 13:13-13:14            LOAD_SMALL_INT           0
+ 13:13-13:14            BUILD_TUPLE              3
+ 13:13-13:14            CALL_INTRINSIC_2         6 (INTRINSIC_SET_TYPEPARAM_OWNER)
```

Reply:
In this case, you can do `PyTuple_CheckExact()` and `PyUnicode_CheckExact`